api/v1alpha2/domachine_types.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	capierrors "sigs.k8s.io/cluster-api/errors"
+)
+
+const (
+	// MachineFinalizer allows ReconcileDOMachine to clean up DigitalOcean resources associated with DOMachine before
+	// removing it from the apiserver.
+	MachineFinalizer = "domachine.infrastructure.cluster.x-k8s.io"
+)
+
+// DOMachineSpec defines the desired state of DOMachine
+type DOMachineSpec struct {
+	// ProviderID is the unique identifier as specified by the cloud provider.
+	// +optional
+	ProviderID *string `json:"providerID,omitempty"`
+	// Droplet size. It must be known DigitalOcean droplet size. See https://developers.digitalocean.com/documentation/v2/#list-all-sizes
+	Size string `json:"size"`
+	// Droplet image
+	Image intstr.IntOrString `json:"image"`
+	// SSHKeys is the ssh key id or fingerprint to attach in DigitalOcean droplet.
+	// It must be available on DigitalOcean account. See https://developers.digitalocean.com/documentation/v2/#list-all-keys
+	SSHKeys []intstr.IntOrString `json:"sshKeys"`
+	// RootDeviceSize is the size of the root volume in GB.
+	// Defaults to 30.
+	// +optional
+	RootDeviceSize int64 `json:"rootDeviceSize,omitempty"`
```

Comment:
Since Droplets have the predefined disk size which you can't change, what is the purpose/effect of this field?

Reply:
initially i planned to make it as external disk, but i have just realized that we should reconfigure attached volume to be useful. :sweat_smile: 
i'll drop this.